group_project_v2/utils.py

```diff
@@ -269,3 +269,14 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.field._values = self._old_values_value
         return False
+
+
+def add_resource(block, resource_type, path, fragment, via_url=False):
+    if via_url:
+        action = fragment.add_javascript_url if resource_type == 'javascript' else fragment.add_css_url
+        action_parameter = block.runtime.local_resource_url(block, path)
+    else:
+        action = fragment.add_javascript if resource_type == 'javascript' else fragment.add_css
+        action_parameter = loader.load_unicode(path)
+
+    action(action_parameter)
```

Comment:
It's sometimes worthwhile to dispatch based on function parameters when there is a lot of common code.  Hoowever, the only common code in this case is

    action(action_parameter)

It would be more readable, more efficient and more succinct to have four functions instead:

    def add_javascript(fragment, path):
        fragment.add_javascript(loader.load_unicode(path))

    def add_javascript(fragment, path):
        fragment.add_javascript(loader.load_unicode(path))

    def add_javascript_url(fragment, path, block):
        fragment.add_javascript_url(block.runtime.local_resource_url(block, path))

    def add_css_url(fragment, path, block):
        fragment.add_css_url(block.runtime.local_resource_url(block, path))